api/lock.go

```diff
@@ -288,9 +288,19 @@ func (l *Lock) createSession() (string, error) {
 	}
 	id, _, err := session.Create(se, nil)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return id, nil
+
+	l.sessionRenew = make(chan struct{})
+	l.lockSession = id
+	go session.RenewPeriodic(l.opts.SessionTTL, id, nil, l.sessionRenew)
+
+	return func() {
+		if !l.isHeld {
```

Comment:
The returned little def function here is run within the Lock, Unlock, Destroy functions after the defer Unlock(), and since defer is LIFO, it should be safe.